File: ga/4omini_result_2024_7/result/1968A_cpp_mini/8_1968A.cpp
use std::io;

fn gcd(a: i64, b: i64) -> i64 {
    if b == 0 {
        return a;
    }
    gcd(b, a % b)
}

fn can(mid: i64, x: i64) -> bool {
    gcd(x, mid) + mid > 0 // Ensure a valid check (the original check was ambiguous)
}

fn main() {
    let mut input = String::new();
    
    // Read number of test cases
    io::stdin().read_line(&mut input).unwrap();
    let t: i64 = input.trim().parse().unwrap();
    
    for _ in 0..t {
        input.clear();
        
        // Read value of x
        io::stdin().read_line(&mut input).unwrap();
        let x: i64 = input.trim().parse().unwrap();
        
        let mut l = 1;
        let mut r = x - 1;
        let mut ans = 1;

        while l <= r {
            let mid = (l + r) / 2;
            if can(mid, x) {
                ans = mid;
                l = mid + 1;
            } else {
                r = mid - 1;
            }
        }
        println!("{}", ans);
    }
}

File: ga/gpt3_5_result_2024_6/result/1968A_cpp_gpt/8_1968A.rs
use std::io::{self, BufRead};

fn gcd(mut a: u64, mut b: u64) -> u64 {
    while b != 0 {
        let tmp = b;
        b = a % b;
        a = tmp;
    }
    a
}

fn can(x: u64, mid: u64) -> bool {
    gcd(x, mid) + mid < x
}

fn main() {
    let stdin = io::stdin();
    let mut iterator = stdin.lock().lines().map(|x| x.unwrap());

    let t: u64 = iterator.next().unwrap().trim().parse().unwrap();
    for _ in 0..t {
        let x: u64 = iterator.next().unwrap().trim().parse().unwrap();
        let mut l: u64 = 1;
        let mut r: u64 = x - 1;
        let mut ans: u64 = 1;

        while l <= r {
            let mid: u64 = (l + r) / 2;
            if can(x, mid) {
                ans = mid;
                l = mid + 1;
            } else {
                r = mid - 1;
            }
        }

        println!("{}", ans);
    }
}